cirq-ionq/cirq_ionq/service.py

```diff
@@ -104,14 +107,36 @@ def run(
         Returns:
             A `cirq.Result` for running the circuit.
         """
+        transpiled_circuit = self.to_native_gates(circuit)
         resolved_circuit = cirq.resolve_parameters(circuit, param_resolver)
-        result = self.create_job(resolved_circuit, repetitions, name, target).results()
+        result = self.create_job(
+            resolved_circuit, repetitions, name, target).results()
         if isinstance(result, results.QPUResult):
             return result.to_cirq_result(params=cirq.ParamResolver(param_resolver))
         # pylint: disable=unexpected-keyword-arg
         return result.to_cirq_result(params=cirq.ParamResolver(param_resolver), seed=seed)
         # pylint: enable=unexpected-keyword-arg
 
+    def to_native_gates(self, circuit: cirq.Circuit):
+        """Converts a `cirq.Circuit` to IonQ native gates.
+
+        Args:
+            circuit: The circuit to transpile.
+
+        Returns:
+            A `cirq.Circuit` that uses native gates.
+        """
+        # Transpile the circuit into the native basis
+        native_circuit = cirq.optimize_for_target_gateset(
+            circuit,
+            gateset=cirq.Gateset.from_ops(
+                GPIGate(),
+                GPI2Gate(),
+                MSGate()
+            )
```

Comment:
The gateset here should be a `CompilationTargetGateset` so that it can tell the transformer how to decompose arbitrary operation to the specific gateset. The transformer would not work out of the box for any arbitrary gateset, like the one attempted here. 

See https://github.com/quantumlib/Cirq/tree/master/cirq-core/cirq/transformers/target_gatesets for examples on how to write a CompilationTargetGateset